### ISSUE TITLE
Doesn't compile with GHC 9.2.3

### ISSUE THREAD

**Issue by Mikolaj** [haskell/stylish-haskell#405]
Fails with

```
~$ cabal install stylish-haskell
Resolving dependencies...
Build profile: -w ghc-9.2.3 -O1
In order, the following will be built (use -v for more details):
 - stylish-haskell-0.14.2.0 (lib) (requires build)
 - stylish-haskell-0.14.2.0 (exe:stylish-haskell) (requires build)
Starting     stylish-haskell-0.14.2.0 (lib)
Building     stylish-haskell-0.14.2.0 (lib)

Failed to build stylish-haskell-0.14.2.0.
Build log (
/home/mikolaj/.cabal/logs/ghc-9.2.3/stylish-haskell-0.14.2.0-b54ac8d49070c15bc96bb1ea4cb38bcc1ab66990033c8c3a3f41c50a1f1f94c5.log
):
Configuring library for stylish-haskell-0.14.2.0..
Preprocessing library for stylish-haskell-0.14.2.0..
Building library for stylish-haskell-0.14.2.0..
[ 1 of 26] Compiling Language.Haskell.Stylish.Block ( lib/Language/Haskell/Stylish/Block.hs, dist/build/Language/Haskell/Stylish/Block.o, dist/build/Language/Haskell/Stylish/Block.dyn_o )
[ 2 of 26] Compiling Language.Haskell.Stylish.Config.Internal ( lib/Language/Haskell/Stylish/Config/Internal.hs, dist/build/Language/Haskell/Stylish/Config/Internal.o, dist/build/Language/Haskell/Stylish/Config/Internal.dyn_o )
[ 3 of 26] Compiling Language.Haskell.Stylish.Editor ( lib/Language/Haskell/Stylish/Editor.hs, dist/build/Language/Haskell/Stylish/Editor.o, dist/build/Language/Haskell/Stylish/Editor.dyn_o )
[ 4 of 26] Compiling Language.Haskell.Stylish.GHC ( lib/Language/Haskell/Stylish/GHC.hs, dist/build/Language/Haskell/Stylish/GHC.o, dist/build/Language/Haskell/Stylish/GHC.dyn_o )

lib/Language/Haskell/Stylish/GHC.hs:71:32: error:
    • Couldn't match expected type ‘GHC.Settings’
                  with actual type ‘ghc-lib-parser-9.2.3.20220527:GHC.Settings.Settings’
      NB: ‘GHC.Settings’
            is defined in ‘GHC.Settings’ in package ‘ghc-9.2.3’
          ‘ghc-lib-parser-9.2.3.20220527:GHC.Settings.Settings’
            is defined in ‘GHC.Settings’
                in package ‘ghc-lib-parser-9.2.3.20220527’
    • In the first argument of ‘defaultDynFlags’, namely
        ‘GHCEx.fakeSettings’
      In the expression:
        defaultDynFlags GHCEx.fakeSettings GHCEx.fakeLlvmConfig
      In an equation for ‘baseDynFlags’:
          baseDynFlags
            = defaultDynFlags GHCEx.fakeSettings GHCEx.fakeLlvmConfig
   |
71 | baseDynFlags = defaultDynFlags GHCEx.fakeSettings GHCEx.fakeLlvmConfig
   |                                ^^^^^^^^^^^^^^^^^^

lib/Language/Haskell/Stylish/GHC.hs:71:51: error:
    • Couldn't match expected type ‘GHC.LlvmConfig’
                  with actual type ‘ghc-lib-parser-9.2.3.20220527:GHC.Driver.Session.LlvmConfig’
      NB: ‘GHC.LlvmConfig’
            is defined in ‘GHC.Driver.Session’ in package ‘ghc-9.2.3’
          ‘ghc-lib-parser-9.2.3.20220527:GHC.Driver.Session.LlvmConfig’
            is defined in ‘GHC.Driver.Session’
                in package ‘ghc-lib-parser-9.2.3.20220527’
    • In the second argument of ‘defaultDynFlags’, namely
        ‘GHCEx.fakeLlvmConfig’
      In the expression:
        defaultDynFlags GHCEx.fakeSettings GHCEx.fakeLlvmConfig
      In an equation for ‘baseDynFlags’:
          baseDynFlags
            = defaultDynFlags GHCEx.fakeSettings GHCEx.fakeLlvmConfig
   |
71 | baseDynFlags = defaultDynFlags GHCEx.fakeSettings GHCEx.fakeLlvmConfig
   |                                                   ^^^^^^^^^^^^^^^^^^^^
Error: cabal: Failed to build stylish-haskell-0.14.2.0 (which is required by
exe:stylish-haskell from stylish-haskell-0.14.2.0). See the build log above
for details.
```

**Comment by teto** [comment]
I think that is blocking the haskell-language-server 923 release, (at least when I tried to package the plugin-enabled version in nix).

**Comment by kokobd** [comment]
Enabling cabal flag `ghc-lib` for package `stylish-haskell` should help. Example:
```
cabal install --constraint "stylish-haskell +ghc-lib" stylish-haskell
```

**Comment by Mikolaj** [comment]
Confirmed. So, I guess, the flag should be conditionally enabled in .cabal? Or marked as not manual, so that cabal can try it out (I haven't tested that variant)?

**Comment by Mikolaj** [comment]
Actually, the latter won't work, because it's a compilation failure, not a build plan failure.

**Comment by teto** [comment]
master works on nixos Thanks !  the service would always crash with that non-helpful error.
Does it warrant a new release ? 

NB: there were many warnings during the builds, here are some if of interest
```
/nix/store/x35d72904sc58is4j4lsgzh3jkbidig4-glibc-2.34-210-dev/include/bits/string_fortified.h:95:10: warning: '__builtin_strncpy' output may be truncated copying 49 bytes from a string of length 199 [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation-Wstringop-truncation8;;]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   96 |                                   __glibc_objsize (__dest));
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
 CC util_list.o
 CC util_request.o
 CC util_salloc.o
 CC util_sql_split.o
 CC util_string.o
 CC ws_delete.o
 CC ws_insert.o
ws_insert.c: In function 'ws_insert_step3':
ws_insert.c:368:16: warning: unused variable 'int_where_len' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-variable-Wunused-variable8;;]
  368 |         size_t int_where_len = 0;
      |                ^~~~~~~~~~~~~
ws_insert.c:365:16: warning: unused variable 'int_len_content' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-variable-Wunused-variable8;;]
  365 |         size_t int_len_content;
      |                ^~~~~~~~~~~~~~~
ws_insert.c: In function 'ws_insert_step4':
ws_insert.c:468:16: warning: unused variable 'int_where_len' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-variable-Wunused-variable8;;]
  468 |         size_t int_where_len = 0;
      |                ^~~~~~~~~~~~~
ws_insert.c:467:16: warning: unused variable 'int_sql_len' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-variable-Wunused-variable8;;]
  467 |         size_t int_sql_len = 0;
      |                ^~~~~~~~~~~
ws_insert.c:465:16: warning: unused variable 'int_len_content' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-variable-Wunused-variable8;;]
  465 |         size_t int_len_content;
      |                ^~~~~~~~~~~~~~~
ws_insert.c:464:39: warning: unused variable 'client_insert' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-variable-Wunused-variable8;;]
  464 |         struct sock_ev_client_insert *client_insert = (struct sock_ev_client_insert *)(client_request->client_request_data);
      |                                       ^~~~~~~~~~~~~
ws_insert.c: In function 'ws_insert_view_step4':
ws_insert.c:826:16: warning: unused variable 'int_temp_len' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-variable-Wunused-variable8;;]
  826 |         size_t int_temp_len = 0;
      |                ^~~~~~~~~~~~
ws_insert.c: In function 'ws_insert_table_step2':
ws_insert.c:984:16: warning: unused variable 'int_where_len' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-variable-Wunused-variable8;;]
  984 |         size_t int_where_len = 0;
      |                ^~~~~~~~~~~~~
ws_insert.c:981:16: warning: unused variable 'int_len_content' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-variable-Wunused-variable8;;]
  981 |         size_t int_len_content;
      |                ^~~~~~~~~~~~~~~
ws_insert.c: In function 'ws_insert_table_step3':
ws_insert.c:1086:16: warning: unused variable 'int_temp_len' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-variable-Wunused-variable8;;]
 1086 |         size_t int_temp_len = 0;
 ```

**Comment by sloorush** [comment]
Same issue with `ghc 9.2.2` as well

> Enabling cabal flag `ghc-lib` for package `stylish-haskell` should help. Example:
> 
> ```
> cabal install --constraint "stylish-haskell +ghc-lib" stylish-haskell
> ```

This works though

**Comment by peterbecich** [comment]
This error also occurs when `stylish-haskell` is used as a `haskell.nix` `shell.tool`: https://input-output-hk.github.io/haskell.nix/tutorials/getting-started-flakes.html#scaffolding

example: https://github.com/peterbecich/halogen-chess/blob/cc1f5f6328d4571e1b0331ab58c88b6596015cff/flake.nix#L201-L210

A.f.a.i.k. `haskell.nix` builds these tools directly from Hackage, I don't know where the solution above `cabal install --constraint "stylish-haskell +ghc-lib" stylish-haskell` would be applied. I don't know if the Nix configuration here is used for `shell.tool`: https://github.com/haskell/stylish-haskell/blob/main/default.nix

**Comment by andreasabel** [comment]
Same problem with GHC 9.2.5 and `stylish-haskell-0.14.3.0`.

**Comment by peterbecich** [comment]
Can this be closed? It appears the GHC 9.2 build and tests succeed: https://github.com/haskell/stylish-haskell/actions/runs/4339790807/jobs/7577708631

**Comment by Mikolaj** [comment]
Yes, it works great, thank you.
